### ISSUE TITLE
Destroy veth device on stop

### ISSUE THREAD

**Issue by stgraber** [lxc/lxc#1253]
This relates to https://github.com/lxc/lxd/issues/2439

To avoid the race condition where the kernel takes too long to expire the network namespace and possibly keeps it alive indefinitely, we should destroy the host side of the veth pair on exit.

**Comment by stgraber** [comment]
So @brauner tells me that we do have code for that, now the question is whether said code is run properly and if it is, why is the kernel telling us the device was deleted when it's still there.

**Comment by brauner** [comment]
It seems we're reliably hitting a codepath for root-owned unprivileged containers that deletes the network interfaces on container exit.

**Comment by brauner** [comment]
So let me rephrase: We explicitly remove the network devices when we haven't moved it to the network namespace. In the case where we explicitly moved it to the network namespace we do not. So I'd suggest that on shutdown we attach to the network namespace and try to remove the network device explicitly.

**Comment by stgraber** [comment]
Hmm, you'd think that deleting the host side of the veth pair would be enough...
